pkg/reconciler/buildrun/resources/sources/http.go

```diff
@@ -25,7 +25,7 @@ func AppendHTTPStep(
 	// HTTP is done currently all in a single step, see if there is already one
 	httpStep := findExistingHTTPSourcesStep(taskSpec)
 	if httpStep != nil {
-		httpStep.Container.Args[3] = fmt.Sprintf("%s ; wget \"%s\"", httpStep.Container.Args[3], source.URL)
+		httpStep.Container.Args[3] = fmt.Sprintf("%s ; wget %q", httpStep.Container.Args[3], source.URL)
```

Comment:
Fix is nice. Potentially even fixing a code injection. But why not in a separate PR? Simlar concern applies to your restructuring in the build reconciler. This is not nice for reviewers.

Reply:
No problem creating a spin-off based on this commit.